infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/NegotiatingState.java

```diff
@@ -64,8 +63,8 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                 fsm.updateDisconnectedNodes(nodeIdDown);
 
                 // If the leader is the node that become unavailable, verify new leader and attempt to reconnect.
-                if (leaderNodeId.isPresent() && leaderNodeId.get().equals(nodeIdDown)) {
-                    leaderNodeId = Optional.empty();
+                if (fsm.getRemoteLeaderNodeId().isPresent() && fsm.getRemoteLeaderNodeId().get().equals(nodeIdDown)) {
```

Comment:
Can we add a comment on why we shouldn't cache the leader node info since this caused an issue?

Reply:
updated the comment in the previous line to reflect the current behavior.